contrib/gomodule/redigo/redigo.go

```diff
@@ -53,6 +60,14 @@ func parseOptions(options ...interface{}) ([]redis.DialOption, *dialConfig) {
 	return dialOpts, cfg
 }
 
+func wrapConn(c redis.Conn, p *params) redis.Conn {
+	if connWithTimeout, ok := c.(redis.ConnWithTimeout); ok {
+		return ConnWithTimeout{connWithTimeout, p}
+	}
+
+	return Conn{c, p}
+}
```

Comment:
I'm slightly confused (but also not entirely familiar with this library, so excuse me if I'm wrong), but wouldn't it have been enough to simply add the two missing methods to `Conn`? https://godoc.org/github.com/gomodule/redigo/redis#ConnWithTimeout

Reply:
I'm thinking this is the only change needed:
```go
func (tc Conn) DoWithTimeout(readTimeout time.Duration, commandName string, args ...interface{}) (reply interface{}, err error) {
    cwt, ok := tc.Conn.(redis.ConnWithTimeout)
    if !ok {
        return
    }
    // .. continue
```